### PR TITLE
chore(ci): run e2e tests in groups COMPASS-7853

### PR DESCRIPTION
**Pull request by mcasimir**
Runs e2e tests in groups of 3. 

Additionally the PR has few drive-bys and follow ups from #5898.

the group number is configurable, a follow up to split e2e tests in more files and prefix them with "numbers" so to balance  the tasks will be necessary for this to be effective. n = 10 reduced the execution time by half (though it also uses 10 times the runners)

### CODE REVIEW

File: .evergreen/buildvariants-and-tasks.in.yml [hunk 1]
```diff
@@ -53,31 +57,31 @@ const PACKAGE_BUILD_VARIANTS = [
 
 const TEST_PACKAGED_APP_BUILD_VARIANTS = [
   {
-    name: 'test-packaged-app-ubuntu',
+    name: 'test-server-ubuntu',
```
Comment (gribnoysup) [inline .evergreen/buildvariants-and-tasks.in.yml:60]: Those runs look weird in CI, only 2 tests are running in every variant:

image

Comment (mcasimir) [reply .evergreen/buildvariants-and-tasks.in.yml:60]: that is weird yeah

File: packages/compass-e2e-tests/index.ts [hunk 2]
```diff
@@ -252,10 +252,23 @@ async function main() {
...
+    })
+  ).filter((value, index, array) => {
+    const testsPerGroup = Math.ceil(array.length / e2eTestGroupsAmount);
+    const minGroupIndex = (e2eTestGroup - 1) * testsPerGroup;
+    const maxGroupIndex = minGroupIndex + testsPerGroup - 1;
+
+    return index >= minGroupIndex && index <= maxGroupIndex;
   });
 
+  console.log(rawTests);
```
Comment (mcasimir) [inline packages/compass-e2e-tests/index.ts:270]:
```suggestion
  console.log('test files:', rawTests);
```

**Review by gribnoysup — APPROVED** [review]

**Review by lerouxb — APPROVED** [review]

File: packages/compass-e2e-tests/index.ts (hunk 2 above)
Comment (paula-stacho) [inline packages/compass-e2e-tests/index.ts:270]: leftover?

Comment (mcasimir) [reply packages/compass-e2e-tests/index.ts:270]: leftover :)

Comment (mcasimir) [inline packages/compass-e2e-tests/index.ts]:
```suggestion
  console.info('Test files:', rawTests);
```

File: .evergreen/buildvariants-and-tasks.in.yml [hunk 3]
```diff
@@ -442,6 +459,31 @@ for (const serverVersion of SERVER_VERSIONS) { %>
...
+        vars:
+          mongodb_version: latest
+          browser_name: '<%= browser %>'
+          compass_distribution: compass
+          e2e_test_groups: <%= E2E_TEST_GROUPS.length %>
+          e2e_test_group: <%= group.number %>
+          debug: 'compass-e2e-tests*,electron*,hadron*,mongo*'
+  <% } %>
+<% } %>
+
```
Comment (addaleax) [inline .evergreen/buildvariants-and-tasks.in.yml:486]: You're inserting this task in the middle of another task, that's why it's running functions it's not supposed to 🙂

Comment (mcasimir) [reply .evergreen/buildvariants-and-tasks.in.yml:477]: fixed, i hope

**Pull request merged**